sdk/identity/azure-identity/src/workload_identity_credential.cpp

```diff
@@ -70,6 +72,17 @@ WorkloadIdentityCredential::WorkloadIdentityCredential(
               "urn%3Aietf%3Aparams%3Aoauth%3Aclient-assertion-type%3Ajwt-bearer" // cspell:disable-line
               "&client_id=")
         + Url::Encode(clientId);
+
+    IdentityLog::Write(
+        IdentityLog::Level::Informational, GetCredentialName() + " was created successfully.");
+  }
+  else
+  {
+    IdentityLog::Write(
+        IdentityLog::Level::Warning,
+        GetCredentialName()
+            + " was not initialized with underlying credential. Environment variables are not "
```

Comment:
`WorkloadIdentityCredential` is not an "umbrella" credential, such as `EnvironmentCredential`, which can have `ClientSecretCredential` or `ClientCertificateCredential` as underlying credential, therefore WIC does not really have a concept of "underlying credential" (at least not from the customer's perspective).

I recommend to pick the wording from one of the managed identity sources, which are in the similar situation in terms of environment variables that need to be set on the machine for the credential to work.

I.e. something like `"Azure Kubernetes environment is not set up for the " + GetCredentialName() + " credential to work."`

Reply:
I.e. `EnvironmentCredential` does really talk talks about it its "underlying credential" in its further log messages:
```cpp
EXPECT_EQ(log[0].first, Logger::Level::Informational);
EXPECT_EQ(
    log[0].second,
    "Identity: EnvironmentCredential gets created with ClientSecretCredential.");

EXPECT_EQ(log[1].first, Logger::Level::Verbose);
EXPECT_EQ(
    log[1].second,
    "Identity: EnvironmentCredential: 'AZURE_TENANT_ID', 'AZURE_CLIENT_ID', "
    "'AZURE_CLIENT_SECRET', and 'AZURE_AUTHORITY_HOST' environment variables are set, so "
    "ClientSecretCredential with corresponding tenantId, clientId, clientSecret, and "
    "authorityHost gets created.");
```